header.go

```diff
@@ -308,6 +308,10 @@ func fixMangledMediaType(mtype string, sep rune) string {
 		// Only terminate with semicolon if not the last parameter and if it doesn't already have a
 		// semicolon.
 		if i != len(parts)-1 && !strings.HasSuffix(mtype, ";") {
+			// Remove whitespace between parameter=value and ;
+			for strings.HasSuffix(mtype, " ") || strings.HasSuffix(mtype, "\t") {
```

Comment:
https://staticcheck.io/docs/checks#S1017 suggests we replace this check and slice with two calls to strings.TrimSuffix

I also wonder if it would be better to use TrimRight, as your implementation will only remove a single trailing whitespace character.

Reply:
I think TrimRight is a better approach. It removes everything in the cutset. No need for 2 TrimSuffix calls.